feature-utils/poly-analysis/src/analysis.js

```diff
@@ -0,0 +1,112 @@
+import { Telemetry } from "../utils/performance-telemetry.js";
+import { createErrorStatus, createSuccessStatus } from "../utils/status.js";
+
+import InactiveCardsSummary from "./analysis/report/inactive-cards-summary.js";
+
+import MinistoriesStatusAnalysis from "./analysis/report/ministories-status-analysis.js";
+
+export class UnrecognizedData {
+    constructor(analysesResults) {
+        this._activeReportAnalyses = analysesResults
+            .filter(
+                ({ analysis, status }) =>
+                    status.isSuccess &&
+                    analysis.isForDataReport &&
+                    analysis.active
+            )
+            .map(({ analysis }) => analysis);
+
+        const inactiveCardsSummary = new InactiveCardsSummary(analysesResults);
+        if (inactiveCardsSummary.active) {
+            this._activeReportAnalyses.push(inactiveCardsSummary);
+        }
+
+        const statusAnalysis = new MinistoriesStatusAnalysis(analysesResults);
+        if (statusAnalysis.active) {
+            this._activeReportAnalyses.push(statusAnalysis);
+        }
+
+        this.active = this._activeReportAnalyses.length > 0;
+    }
+
+    get reportAnalyses() {
+        return this._activeReportAnalyses;
+    }
+
+    get report() {
+        if (!this.active) {
+            return "No data to report!";
+        }
+        return (
+            this.reportAnalyses.length +
+            " " +
+            (this.reportAnalyses.length > 0 ? "analyses" : "analysis") +
+            "  included in the report"
+        );
+    }
+
+    get jsonReport() {
+        if (!this.active) {
+            return {};
+        }
+
+        const reportAnalyses = this.reportAnalyses.map(
+            (analysis) => analysis.jsonReport
+        );
+
+        return { reportAnalyses_v1: reportAnalyses };
+    }
+}
+
+class AnalysisExecutionResult {
+    constructor(analysis, status, executionTime) {
+        this._analysis = analysis;
+        this._status = status || createSuccessStatus();
+        this._executionTime = executionTime;
+    }
+
+    get analysis() {
+        return this._analysis;
+    }
+
+    get status() {
+        return this._status;
+    }
+
+    get executionTime() {
+        return this._executionTime;
+    }
+
+    get reportJsonData() {
```

Comment:
Disagree that this should eventually be TypeScript, but that has nothing to do with inconsistent naming. I agree this is weird, but I'd rather keep the changes to this minimal for now, given how big this PR is. Though I wouldn't object to calling this `jsonReport` either.

Reply:
Not saying this *should* be TS (although it might help), but TS would allow us to define interfaces that would have to be implemented. There's no equivalent way of enforcing consistency in naming in JS, AFAICT